### PR TITLE
Refactoring Python code to eliminate code smells

### PR DESCRIPTION
**Pull request by ekiwaka**
Fixed a few code smells as suggested by Sonar Lint

Changes proposed in this PR include:
"Remove this commented out code."
-- Link: https://sonarcloud.io/project/issues?open=AYcsbV4WMh9La1tIBkbn&id=PowerGridModel_power-grid-model
-- This issue was fixed in src/power_grid_model/core/power_grid_meta.py
-- Code that was commented out was removed from the function.

"Define a constant instead of duplicating this literal " and " 3 times."
-- Link: https://sonarcloud.io/project/issues?open=AYcsbV5BMh9La1tIBkbq&id=PowerGridModel_power-grid-model
-- This issue was fixed in src/power_grid_model/validation/errors.py
-- A variable named delimiter is added as a class attribute which can then be used by class functions.

"Refactor this function to reduce its Cognitive Complexity from 22 to the 15 allowed."
-- Link: https://sonarcloud.io/project/issues?open=AYmXiBpXOdT128GMx9_u&id=PowerGridModel_power-grid-model
-- This issue was fixed in src/power_grid_model/validation/validation.py
-- Instead of multiple if functions, a dictionary and a for loop were used to iterate through the error data. This way, you're looping through the dictionary component_validators, checking if each component exists in the data dictionary, and if it does, you apply the corresponding validation function to that component's data. This approach makes the code more concise and easier to maintain.

### CODE REVIEW

File: src/power_grid_model/validation/errors.py [hunk 1]
```diff
@@ -49,6 +49,8 @@ class ValidationError(ABC):
 
     _message: str = "An unknown validation error occurred."
 
+    delimiter: str = " and "
```
Comment (mgovers) [inline src/power_grid_model/validation/errors.py:52]: Nitpick: make private

```suggestion
    _delimiter: str = " and "
```

**Review by mgovers — COMMENTED** [review]
Hi @ekiwaka ,

First of all, thank you for your contribution! It excites us that more people are starting to join us. You're actually the first external contributor that makes a code change. Are you ok with us announcing that at the next PGM meet-up?

The proposed changes are looking good. Good design choice on the validators! I only have one small nit-pick remark but it is good to go.

File: src/power_grid_model/validation/validation.py [hunk 2]
```diff
@@ -360,28 +360,23 @@ def validate_values(  # pylint: disable=too-many-branches
...
+        "sym_load": lambda d: validate_generic_load_gen(d, "sym_load"),
+        "sym_gen": lambda d: validate_generic_load_gen(d, "sym_gen"),
+        "asym_load": lambda d: validate_generic_load_gen(d, "asym_load"),
+        "asym_gen": lambda d: validate_generic_load_gen(d, "asym_gen"),
+        "shunt": validate_shunt,
+    }
+
+    for component, validator in component_validators.items():
+        if component in data:
+            errors += validator(data)
```
Comment (mgovers) [inline src/power_grid_model/validation/validation.py:379]: Please remove the trailing whitespace as well

**Comment by Jerry-Jinfeng-Guo** [comment]
@ekiwaka If you could rename the PR to a better descriptive title, I'll further go over the changes and give the approving review once you have processed comments from Marthijn.

File: src/power_grid_model/validation/errors.py (hunk 1 above)
Comment (ekiwaka) [reply src/power_grid_model/validation/errors.py]: Thank you for the nice feedback! Sure, you can announce this in your next meetup!
I have made the requested change and made the delimiter private.

File: src/power_grid_model/validation/validation.py (hunk 2 above)
Comment (ekiwaka) [reply src/power_grid_model/validation/validation.py]: I have removed this trailing whitespace!

**Comment by ekiwaka** [comment]
@Jerry-Jinfeng-Guo I have made the changes requested by @mgovers

**Comment by mgovers** [comment]
@ekiwaka the changes look good. Can you fix the DCO issue? I recommend just doing 2 separate commits with the DCO message, where the second one reverses the first one.

If you really know what you're doing you can also amend the current comment

**Comment by ekiwaka** [comment]
@mgovers done both

**Review by mgovers — APPROVED** [review]

**Comment by mgovers** [comment]
Thank you @ekiwaka . LGTM

**Pull request merged**